fix(OTEL): audit of OTEL docs (#18627)
newrelic/docs-website: src/content/docs/opentelemetry/best-practices/opentelemetry-otlp.mdx

  - /docs/more-integrations/open-source-telemetry-integrations/opentelemetry/best-practices/opentelemetry-otlp
---

[OpenTelemetry Protocol](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md) (OTLP) is a general-purpose telemetry data delivery protocol designed for the OpenTelemetry project. Each OpenTelemetry language SDK provides OTLP exporters, and the OpenTelemetry Collector has OTLP receivers and exporters. Additionally, various tools outside the OpenTelemetry project have added support for OTLP export.

> New Relic supports native OTLP ingest and recommend it as the preferred method for sending OpenTelemetry data to the New Relic platform. This page covers New Relic's OTLP support including configuration, requirements, and recommendations.

Author: ```suggestion
New Relic supports native OTLP ingest and recommends it as the preferred method for sending OpenTelemetry data to the New Relic platform. This page covers New Relic's OTLP support including configuration, requirements, and recommendations.
```

## Before you begin [#before-you-begin]

* If you haven't already done so, sign up for a free [New Relic account](https://newrelic.com/signup).

> * Get your [license key](https://one.newrelic.com/launcher/api-keys-ui.launcher) for the New Relic account to which you want to report data. This license key will be used when [configuring the `api-key` header](#api-key).

Author: ```suggestion
* Get your <InlinePopover type=“license” /> for the New Relic account to which you want to report data. This license key will be used when [configuring the `api-key` header](#api-key).
```

The mechanism to configure attribute limits will vary, but OpenTelemetry language SDKs generally support setting the `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT=4095` and `OTEL_ATTRIBUTE_COUNT_LIMIT=64` environment variables (see [OpenTelemetry docs](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/) for more info).

> If using the Collector, the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor) can be configured to truncate attributes to New Relic limits.

Reviewer: I would write this to avoid the passive voice:
`When using the collector, you can configure the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor) to truncate attributes to New Relic limits.`

> If using the Collector, the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor) can be configured to truncate attributes to New Relic limits.

Author: ```suggestion
When using the collector, you can configure the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor) to truncate attributes to New Relic limits.
```

The mechanism to configure the endpoint will vary, but OpenTelemetry language SDKs generally support setting the `OTEL_EXPORTER_OTLP_COMPRESSION=gzip` environment variable (see [OpenTelemetry docs](https://opentelemetry.io/docs/specs/otel/protocol/exporter/) for more info).

> If using the Collector, `gzip` is the default compression, but `zstd` can be optionally configured.

Reviewer: I would write this to avoid the passive voice:
`If using the Collector, `gzip` is the default compression, but you can optionally configure `zstd`.`

> If using the Collector, `gzip` is the default compression, but `zstd` can be optionally configured.

Author: ```suggestion
If using the Collector, `gzip` is the default compression, but you can optionally configure `zstd`.
```

The mechanism to configure retry will vary. Some OpenTelemetry SDKs may have language specific environment variables (for example [java supports](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure) setting `OTEL_EXPERIMENTAL_EXPORTER_OTLP_RETRY_ENABLED=true`), but there is no general mechanism. Programmatic configuration may be required.

> If using the Collector, the `otlphttpexporter` and `otlpexporter` retry by default. See [exporterhelper](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) for more details.

Reviewer: `exporterhelper`

> If using the Collector, the `otlphttpexporter` and `otlpexporter` retry by default. See [exporterhelper](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) for more details.

Author: ```suggestion
If using the Collector, the `otlphttpexporter` and `otlpexporter` retry by default. See [`exporterhelper`](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) for more details.
```
